cluster-autoscaler/cloudprovider/test/test_cloud_provider.go

```diff
@@ -231,45 +231,51 @@ func (tcp *TestCloudProvider) NewNodeGroupWithId(machineType string, labels map[
 	}, nil
 }
 
-// InsertNodeGroup adds already created node group to test cloud provider.
-func (tcp *TestCloudProvider) InsertNodeGroup(nodeGroup cloudprovider.NodeGroup) {
-	tcp.Lock()
-	defer tcp.Unlock()
-
-	tcp.groups[nodeGroup.Id()] = nodeGroup
-}
-
 // BuildNodeGroup returns a test node group.
-func (tcp *TestCloudProvider) BuildNodeGroup(id string, min, max, size int, autoprovisioned bool, machineType string, opts *config.NodeGroupAutoscalingOptions) *TestNodeGroup {
+func (tcp *TestCloudProvider) BuildNodeGroup(id string, min, max, size int, autoprovisioned bool, machineType string, opts *config.NodeGroupAutoscalingOptions, exists bool) *TestNodeGroup {
```

Comment:
I meant this could be combined further with `NewNodeGroup` and `NewNodeGroupWithId`. All three could be combined into a single function with functional options to be set as needed (akin to https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/utils/test/test_utils.go#L36). That's a bit of extra refactoring, so leaving up to you whether or not to do this in this PR.

Reply:
Couldn't modify `NewNodeGroup` unless I modify the interface `cloudprovider.CloudProvider`, so I kept it as it is and removed `NewNodeGroupWithId` to have only one `BuildNodeGroup`

Reply:
Ah, good point. Thanks for reducing the number of methods!